spec.html

```diff
@@ -28538,135 +28360,96 @@ <h1>Math.imul ( _x_, _y_ )</h1>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ is *+&infin;*, return _n_.
+          1. If _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. Return an implementation-approximated value representing the result of the base 10 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.log2">
         <h1>Math.log2 ( _x_ )</h1>
-        <p>Returns an implementation-approximated value representing the result of the base 2 logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <p>Returns the base 2 logarithm of _x_.</p>
+        <p>When the `Math.log2` method is called with argument _x_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ is *+&infin;*, return _n_.
+          1. If _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. Return an implementation-approximated value representing the result of the base 2 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.max">
         <h1>Math.max ( _value1_, _value2_, ..._values_ )</h1>
         <p>Given zero or more arguments, calls ToNumber on each of the arguments and returns the largest of the resulting values.</p>
-        <ul>
-          <li>
-            If no arguments are given, the result is *-&infin;*.
-          </li>
-          <li>
-            If any value is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            The comparison of values to determine the largest value is done using the Abstract Relational Comparison algorithm except that *+0* is considered to be larger than *-0*.
-          </li>
-        </ul>
+        <p>When the `Math.max` method is called with at least two arguments _value1_ and _value2_ and any number of additional arguments which form the rest parameter ..._values_, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _numbers_ be a List containing _value1_, _value2_, and the elements of _values_ in List order.
+          1. Let _highest_ be *-&infin;*.
+          1. For each element _number_ of _numbers_, do
+            1. Let _n_ be ? ToNumber(_number_).
+            1. If _n_ &gt; _highest_, set _highest_ to _n_.
```

Comment:
We need to explicitly treat `NaN`, `+0`, and `-0` here, I think; it does not make sense to ask if `NaN > 1`, and we have to ensure that `+0` is larger than `-0`.

Reply:
How does this look? Would it be better to break the second link into two separate conditions or somehow group the two conditions joined with "and" for better clarity?

```
            1. If _n_ is *NaN*, return *NaN*.
            1. If _n_ &gt; _highest_ or _n_ is +0 and _highest_ is *-0*, set _highest_ to _n_.
```

Reply:
Maybe this is a bit better

```
            1. If _n_ is *NaN*, return *NaN*.
            1. If _n_ is +0 and _highest_ is *-0* or _n_ &gt; _highest_, set _highest_ to _n_.
```

Reply:
@ryanjduffy Please don't combine things like that. The precedence of and/or in English is ambiguous.

Reply:
:+1: That was my instinct as well. I'll separate them out.

Reply:
I've split them into separate lines. Seems to read better but open to suggestions as well.